lib/squadster/formations/services/create_squad.ex

```diff
@@ -0,0 +1,37 @@
+defmodule Squadster.Formations.Services.CreateSquad do
+  alias Squadster.Repo
+  alias Squadster.Formations.{Squad, SquadMember, SquadRequest}
+
+  def call(args, user) do
+    user
+    |> Repo.preload(:squad_member)
+    |> case do
+      %{squad_member: nil}     -> create(args, user)
+      %{squad_member: _member} -> {:error, "User already has squad"}
+    end
+  end
+
+  defp create(args, user) do
+    args
+    |> Squad.changeset
+    |> Repo.insert
+    |> case do
+      {:error, reason} -> {:error, reason}
+      {:ok, squad} ->
+        squad |> add_commander(user)
+        remove_squad_equest(user)
+        {:ok, squad}
+    end
+  end
+
+  defp add_commander(squad, user) do
```

Comment:
Good catch, but I think pattern matching is better in this case, because
`squad |> add_commander(user)`
looks much more friendly than
`squad.id |> add_commander(user.id)`